src/webgpu/api/operation/device/info.spec.ts

```diff
@@ -0,0 +1,34 @@
+export const description = `
+Tests GPUDevice.adapterInfo member.
+`;
+
+import { Fixture } from '../../../../common/framework/fixture.js';
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { getGPU } from '../../../../common/util/navigator_gpu.js';
+import { assert, objectEquals } from '../../../../common/util/util.js';
+
+export const g = makeTestGroup(Fixture);
+
+g.test('device_adapter_info')
+  .desc(
+    `
+  Test that GPUDevice.adapterInfo matches GPUAdapter.info`
+  )
+  .fn(async t => {
+    const gpu = getGPU(t.rec);
+    const adapter = await gpu.requestAdapter();
+    assert(adapter !== null);
+
+    const device = await t.requestDeviceTracked(adapter);
+    assert(device !== null);
+
+    assert(device.adapterInfo instanceof GPUAdapterInfo);
+    assert(adapter.info instanceof GPUAdapterInfo);
+
+    for (const k of Object.keys(GPUAdapterInfo.prototype)) {
```

Comment:
I think we should actually require `device.adapterInfo === adapter.info`. I've proposed it here:
https://github.com/gpuweb/gpuweb/pull/4809#discussion_r1826388788

Reply:
I'll be waiting for https://github.com/gpuweb/gpuweb/pull/4809#pullrequestreview-2411155432 to be resolved.